classes/config/ConfigParser.inc.php

```diff
@@ -103,6 +103,8 @@ static function &readConfig($file) {
 					} else if (defined($value)) {
 						// The value matches a named constant
 						$value = constant($value);
+					} else if (preg_match('/^[A-Z_]+$/', $value) && getenv($value) !== false) {
+						$value = getenv($value);
 					}
 				}
 
```

Comment:
@alvinchevolleaux I believe that the environment variables are strings only, so the `$value` will get a string value if, for example we want to give it a boolean value. Have you tried that scenario?

Reply:
Another thing that we may consider here could be a way to easily distinguish the environment variables that we are using inside the `config.inc.php`. We could do that by adding a pattern for those like, for example, `environment(DB_USER)`